fairmq/sdk/Topology.h

```diff
@@ -69,6 +69,7 @@ struct DeviceStatus
 };
 
 using TopologyState = std::vector<DeviceStatus>;
+using TopologyStateIndex = std::unordered_map<DDSTask::Id, size_t>; //  task id -> index in the data vector
```

Comment:
Do you really need `size_t` here? Do you expect more than 2^31 elements in the data vector? Use `int` to make buckets smaller (I hope), and thus make it more cache efficient.